src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -441,6 +441,8 @@ private SelectItem GenerateExpandItem(ExpandTermToken tokenIn)
 
             ODataExpandPath pathToNavProp = new ODataExpandPath(pathSoFar);
 
+            IEdmTypeReference elementTypeReference = hasDerivedTypeSegment ? derivedType.ToTypeReference() : null;

```

Comment:
It's null since we can get it from the expanded property in these lines
https://github.com/OData/odata.net/blob/145252226e60b4c9735f1c643fe3e01af039eab0/src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs#L924-L929

Reply:
I see that we are getting the type in SelectExpandBinder, but what is the contract with BindFilter, BindOrderby, etc? These private functions take an elementType which gets passed eventually to CreateBindingState where the logic is to fall back to the navigationSource, if it's not otherwise specified. But it's not clear in the various private methods that the elementType is nullable, and future code could make invalid assumptions about whether it may be null.  For maintainability, maybe at least clarify in the description of the elementType parameter (i.e., line 529) "The target element type if different from the type of the target navigation source."

Reply:
@mikepizzo I have updated the description.